### ISSUE TITLE
Loading plugin in tests throws already loaded error

### ISSUE THREAD

**Issue by ZoFem** [cakephp/cakephp#17993]
### Description

If you used to load plugins in your tests, it now throws an error `Possibly related to Cake\Core\Exception\CakeException: "Plugin named X is already loaded"`;

This is my code (in my controller test): 
```php
public function setUp(): void
{
    parent::setUp();
    $this->loadRoutes();
    $this->loadPlugins(['X']);
}
``` 

In the migration guide it says:
- PluginCollection::addPlugin() now throws an exception if a plugin of the same name is already added.
- TestCase::loadPlugins() will now clear out any previously loaded plugins. So you must specify all plugins required for any subsequent tests.

So it seems the TestCase doesn't actually clear out previously loaded plugins. If I remove the newly added line in the PluginCollector in `PluginCollection::addPlugin()` the errors go away.

I tried clearing the plugin manually like this:
```php
$this->clearPlugins();
$this->removePlugins(['X']);
```
But that didn't work either.

### CakePHP Version

5.1.1

### PHP Version

8.2

**Comment by dereuromark** [comment]
Did you also already load plugins in the bootstrap process.
Then you probably run into this as it loads them twice. 
I had similar issues.
I just needed to make sure they are either loaded in bootstrap or on demand inside the tests.

**Comment by ZoFem** [comment]
We load the plugin in the `config/plugins.php` file, as specified in the documentation. 
We haven't added anything to the bootstrap process. 
I tried loading the plugin in the `Application/bootstrap` process, in the `tests/bootstrap.php` file (and tried including the `plugins.php` file there), but couldn't find the right syntax or get that to work.

**Comment by dereuromark** [comment]
Do you still have this issue or did you solve it?
I dont have them in my apps for example, even though I use plenty of plugins (see IdeHelper plugin etc).

**Comment by ZoFem** [comment]
Yes, still having this issue unfortunately. It only seems to happen with specific plugins or own created plugins. We also use IdeHelper for example, but don't get the error on that

**Comment by dereuromark** [comment]
Anything else to be done here?
In my apps I dont get this issue.

**Comment by markstory** [comment]
Closing for now. If there is any follow up we can reopen.
